program-runtime/src/stable_log.rs

```diff
@@ -100,10 +100,11 @@ pub fn program_success(log_collector: &Option<Rc<RefCell<LogCollector>>>, progra
 /// ```notrust
 /// "Program <address> failed: <program error details>"
 /// ```
+#[allow(clippy::borrowed_box)]
 pub fn program_failure(
     log_collector: &Option<Rc<RefCell<LogCollector>>>,
     program_id: &Pubkey,
-    err: &InstructionError,
+    err: &Box<dyn std::error::Error>,
```

Comment:
This should be &dyn Error and then you &* at the call site

Reply:
Tried that already and it complains that the size of `dyn std::error::Error` is not known at compile time.
But I found a different solution: Use `.as_ref()` at call site.

Reply:
as_ref returns &T which is also what deref() returns for Box, so &* should work? And it's the common way to box -> dyn

Reply:
https://buildkite.com/solana-labs/solana/builds/92561#0187135a-5f5c-46a8-a659-c37e9d2e112e